src/rabbit_direct.erl

```diff
@@ -101,6 +105,45 @@ connect(Creds, VHost, Protocol, Pid, Infos) ->
         false -> {error, broker_not_found_on_node}
     end.
 
+extract_extra_auth_props(Creds, VHost, Pid, Infos) ->
+    case extract_protocol(Infos) of
+        undefined ->
+            [];
+        Protocol ->
+            maybe_call_connection_info_module(Protocol, Creds, VHost, Pid, Infos)
+    end.
+
+extract_protocol(Infos) ->
+    case proplists:get_value(protocol, Infos, undefined) of
```

Comment:
Sorry, I don't get it. Here we get the protocol on the fly from the connection info (e.g. `{'MQTT', '3.1.1'}` and then build the module name from it (e.g. `rabbit_mqtt_connection_info`) and check if the module exists.